alpha/scripts/utils/restoreMetadataObjects.php

```diff
@@ -0,0 +1,125 @@
+<?php
+ini_set('memory_limit','1024M');
+if ($argc < 3)
+{
+	echo PHP_EOL . ' ---- Restore Deleted Metadata Profile && Objects ---- ' . PHP_EOL;
+	echo ' Execute: php ' . $argv[0] . ' partnerId, metadata profile ID, [ /path/to/metadata object IDs list || metadataObjectId_1,metadataObjectId_2,.. || metadataObject ] [realrun / dryrun]' . PHP_EOL;
+	die(' Error: missing metadataObjectIds file, csv or single entry ' . PHP_EOL . PHP_EOL);
+}
+
+$partnerId = $argv[1];
+$metadataProfileId = $argv[2];
+$metadataObjectIds = explode(',', $argv[3]);
+$totalMetadataObjectIds = count($metadataObjectIds);
+
+
+require("/opt/kaltura/app/alpha/scripts/bootstrap.php");
+
+$dryRun = true;
+if (isset($argv[3]) && $argv[3] == 'realrun')
+{
+	$dryRun = false;
+}
+
+KalturaStatement::setDryRun($dryRun);
+KalturaLog::info($dryRun ? 'DRY RUN' : 'REAL RUN');
+
+$mdVersion = restoreMDProfile($metadataProfileId ,$partnerId);
+restoreMDObjects($metadataObjectIds, $mdVersion, $partnerId);
+
+function restoreMDProfile($metadataProfileId ,$partnerId)
+{
+	MetadataProfilePeer::setUseCriteriaFilter(false);
+	$metadataProfile = MetadataProfilePeer::retrieveByPK($metadataProfileId);
+	MetadataProfilePeer::setUseCriteriaFilter(true);
+
+	if (!$metadataProfile)
+	{
+		KalturaLog::debug("ERR1 - metadata profile id " . $metadataProfileId . " not found");
+		exit();
+	}
+
+	$mdVersion = $metadataProfile->getVersion();
+	$fileSyncs = getFileSyncs($metadataProfileId, $mdVersion, $partnerId);
+
+	foreach ($fileSyncs as $fileSync)
+	{
+		$fileSyncId = $fileSync->getId();
+		$dc = $fileSync->getDc();
+		KalturaLog::debug('Saving file sync ID: ' . $fileSyncId . ' - version - ' . $mdVersion . ' - dc - ' . $dc);
+		$fileSync->setStatus(2);
+		$fileSync->save();
+	}
+
+	KalturaLog::debug('Saving metadata profile ID: ' . $metadataProfileId);
+	$metadataProfile->setStatus(1);
+	$metadataProfile->save();
+
+	return $mdVersion;
+}
+
+function restoreMDObjects($metadataObjectIds, $mdVersion, $partnerId)
+{
+	foreach ($metadataObjectIds as $metadataObjectId)
+	{
+		$metadataObjectId = trim($metadataObjectId);
+		KalturaLog::debug("=== Metadata Object Id: " . $metadataObjectId);
+		MetadataPeer::setUseCriteriaFilter(false);
+		$metadataObject = MetadataPeer::retrieveByPK($metadataObjectId);
+		MetadataPeer::setUseCriteriaFilter(true);
+
+		/* @var $metadataObjectId metadata */
+
+		if (!$metadataObjectId)
+		{
+			KalturaLog::debug("ERR3 - metadata object id " . $metadataObjectId . " not found");
+			continue;
+		}
+
+		KalturaLog::debug('=== Restoring metadata object id [' . $metadataObject->getID() . ']');
+		$mdOProfileVersion = $metadataObject->getMetadataProfileVersion();
+
+		if ($mdOProfileVersion <> $mdVersion)
```

Comment:
I would put this after checking the metadataObject status.
No need to print errors for versions difference, if the status isn't deleted